packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -265,7 +265,7 @@ export class OdspDocumentServiceFactoryCore
 
 	protected async createDocumentServiceCore(
 		resolvedUrl: IResolvedUrl,
-		odspLogger: TelemetryLogger,
+		odspLogger: ITelemetryLogger,
```

Comment:
Shouldn't this (and the corresponding in `localOdspDocumentServiceFactory.ts`) be taking `ITelemetryBaseLogger` and using `createChildLogger` where it needs the extra APIs? Seems like that might propagate a bit so maybe you tried it and it didn't pan out?

Reply:
that is one path we could take. I tried to keep the change pretty minimal, and i don't want createChildLogger being called too much as it does do a bunch of object allocations.  so I've only updated it on the constructor path for prod code.  There is more cleanup we could do here with all the logger interfaces, but i don't think i'll do that, since there work to finish the ext logger is still outstanding. After this we at least don't expose our implementations.